Mathlib/Analysis/SpecialFunctions/Trigonometric/Basic.lean

```diff
@@ -795,6 +796,44 @@ theorem sin_pi_div_three : sin (π / 3) = √3 / 2 := by
   congr
   ring
 
+theorem four_mul_sq_cos_pi_div_five_sub_two_mul_sub_one_eq_zero :
+    letI c := cos (π / 5)
+    4 * c ^ 2 - 2 * c - 1 = 0 := by
```

Comment:
This is one of those completely unreadable theorem statements. I think rephrasing it might be in order, something like:
```suggestion
theorem Polynomial.isRoot_cos_pi_div_five :
    (4 • X ^ 2 - 2 • X - C 1 : ℝ[X]).IsRoot (cos (π / 5)) := by
```
or whatever is the correct notation for the polynomial.

Reply:
I admit that it's easy to lose perspective about one's own work, but is it really so bad? I mean, it's just a quadratic equation :)

Personally I find `4 * c ^ 2 - 2 * c - 1 = 0` much easier to read than `(4 • X ^ 2 - 2 • X - C 1 : ℝ[X]).IsRoot (cos (π / 5))` but in any case I'd argue we should try to optimise for a statement which is likely to be most convenient to the downstream user. Would you feel differently if the `letI` did not appear? I.e., if it read:
```lean
    4 * cos (π / 5) ^ 2 - 2 * cos (π / 5) - 1 = 0 := by
```

Separately, I have some reservations about using `Polynomial`. It's only a rule of thumb but I feel like the job of `Polynomial` is really to support the study of the type of all polynomials, rather than as the encoding for every expression of degree 2 or more than crops up.

Having said all that, I don't have especially strong feelings here and I will happily make the change if you still feel the same; I just wanted to share my thoughts.

Reply:
> I admit that it's easy to lose perspective about one's own work, but is it really so bad? I mean, it's just a quadratic equation :)

Sorry, I misspoke, but it was exacerbated by the fact that I also rephrased the statement. What I meant to say was:

> This is one of those completely unreadable theorem ~~statements~~ *names*.

As far as the actual *statement* goes, I liked the `letI`. As for whether we have the `Polynomial` version or yours, let's have our cake and eat it too? We can have both!

I do see your point about `Polynomial`, but I think it's okay to use it for both purposes.

Reply:
Ah right, it was the name to which you were objecting.

I hated it too so I've accepted your permission to deviate from standard naming and used `quadratic_root_cos_pi_div_five`.

Also added the `Polynomial` version.